src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1009,9 +1009,7 @@ private IEnumerable<ITypeSymbol> InferTypeInCatchFilterClause(CatchFilterClauseS
                 var rightTypes = GetTypes(coalesceExpression.Right);
                 if (!rightTypes.Any())
                 {
-                    return SpecializedCollections.SingletonEnumerable(
-                        this.Compilation.GetSpecialType(SpecialType.System_Nullable_T)
-                            .Construct(Compilation.GetSpecialType(SpecialType.System_Object)));
```

Comment:
Yeah.  I was expecting some weird confluence of events to have causes this.  Little did i expect :)